cli/compose/types/types.go

```diff
@@ -408,6 +408,11 @@ type FileReferenceConfig struct {
 	UID    string  `yaml:",omitempty" json:"uid,omitempty"`
 	GID    string  `yaml:",omitempty" json:"gid,omitempty"`
 	Mode   *uint32 `yaml:",omitempty" json:"mode,omitempty"`
+	// Runtime is set `true` if reference isn't actually used as a File in
+	// the service. Specifically, this currently only indicates that the Config
+	// is used as a CredentialSpec. If this field is set `true`, then all other
+	// fields besides `Source` must be empty.
+	Runtime bool `yaml:",omitempty" json:"runtime,omitempty"`
```

Comment:
Can we deduct this from the `Source` value? (IIUC, it will be prefixed with `config://` ?)

Reply:
No. The string `config://` should never actually be in the compose code, I think.

We could theoretically determine that a config is a `Runtime` config by looking for the absence of all other fields, but I usually think it's better to affirmatively determine something rather than make assumptions from the absence.

It would be cleaner to have a whole new type besides `FileReferenceConfig` but doing so would be more trouble than it's worth, IMHO.

Reply:
Would it make sense to use a string instead of a bool?
I'd like to avoid a situation where we are adding multiple bools here.

Reply:
A string representing what? The target, file vs runtime?

Reply:
One thing to take note is that this type (`FileReferenceConfig`) is used both for configs and secrets, but we'll only use the "runtime" option for configs; so we might have to create a separate type for configs?

Thinking (UX-wise); in order to use a config as credential-spec, I would have to;

1. define the Config

      ```yaml
      version: "3.8"

      configs:
        my_credentials_spec:
          file: ./credentials-spec-file.json
      ```

2. add the Config to the Configs section of the compose file (I guess this can _only_ be with [the long syntax](https://docs.docker.com/compose/compose-file/#long-syntax), because I'd have to specify the `runtime` option);

      ```yaml
      version: "3.8"
      services:
        myservice:
          image: myimage:latest
          configs:
            - source: my_credentials_spec
              runtime: true

      configs:
        my_credentials_spec:
          file: ./credentials-spec-file.json
      ```

3. configure the service to use the config as source for the credentials-spec;

      ```yaml
      version: "3.8"
      services:
        myservice:
          image: myimage:latest
          credential_spec:
            config: my_credentials_spec
          configs:
            - source: my_credentials_spec
              runtime: true

      configs:
        my_credentials_spec:
          file: ./my-credential-spec.json
      ```

That's a lot of duplication; can we somehow skip `step 2.` ? i.e., if `credential_spec` uses `config: <some name>`, we already know that we want to use a config as source. That config must be defined in the compose-file, but we could attach it automatically to the service (if needed) (or can we do that step even "behind the scenes", daemon-side?)

Reply:
Sure, why not.